app/modules/sightings/resources.py

```diff
@@ -57,21 +80,182 @@ def get(self, args):
+
+        arefs_found = {}
+        i = 0
+        while i < len(data['encounters']):
+            enc_data = data['encounters'][i]
+            # TODO handle regular .assets flavor
+            # TODO genericize this across here and encounters
+            if 'assetReferences' in enc_data and isinstance(
+                enc_data['assetReferences'], list
+            ):
+                enc_data['_paths_wanted'] = []
+                for aref in enc_data['assetReferences']:
+                    if (
+                        not isinstance(aref, dict)
+                        or 'path' not in aref
+                        or 'transactionId' not in aref
+                    ):
+                        log.error(
+                            'Sighting.post malformed assetReferences data: %r' % (aref)
+                        )
+                        _cleanup_post_and_abort(
+                            None, None, 'Malformed assetReferences data'
+                        )
+                    enc_data['_paths_wanted'].append(aref['path'])
+                    key = ':'.join((aref['transactionId'], aref['path']))
+                    if key in arefs_found:
+                        arefs_found[key]['encs'].append(i)
+                    else:
+                        arefs_found[key] = aref.copy()
+                        arefs_found[key]['encs'] = [i]
+            i += 1
+
+        # now we have a mapping of encounters to assetReferences... so we try to create the submission + assets
+        #   first we make sure we have the files we need so we can abort if not (they may fail later for other reasons)
+        from app.extensions.tus import tus_upload_dir
+
+        transaction_id = None  # see note above about single common transaction id
+        for key in arefs_found:
+            aref = arefs_found[key]
+            transaction_id = aref['transactionId']
+            file_path = os.path.join(
+                tus_upload_dir(current_app, transaction_id=transaction_id),
+                aref['path'],
+            )
+            try:
+                sz = os.path.getsize(file_path)  # 2for1
+            except OSError as err:
+                log.error(
+                    'Sighting.post OSError %r assetReferences data: %r' % (err, aref)
+                )
+                _cleanup_post_and_abort(None, None, 'File not found: ' + str(aref))
+            if sz < 1:
+                log.error(
+                    'Sighting.post zero-size file for assetReferences data: %r' % (aref)
+                )
+                _cleanup_post_and_abort(None, None, 'File empty: ' + str(aref))
+
+        # files seem to exist in uploads dir, so lets move on
+        from app.modules.submissions.models import Submission
+
+        pub = False
+        owner_guid = None
+        if current_user is not None and not current_user.is_anonymous:
+            owner_guid = current_user.guid
+            pub = True
```

Comment:
Is this pub logic the right way around? I thought anonymous users created public Sightings and real users created non public sightings?

Reply:
you are 100% right!  fixed.  weirdly i have the logic _correct_ in the encounter analog; which i swear is where i copied this from to start with.  who knows.  late nights?